### PR TITLE
Basic Authentication Integrated

### PR DESCRIPTION
**Pull request by BinaryDevil**

### CODE REVIEW

File: src/main/java/org/openbmp/db_rest/helpers/AuthenticationService.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2014-2015 Cisco Systems, Inc. and others.  All rights reserved.
```
Comment (TimEvens) [inline src/main/java/org/openbmp/db_rest/helpers/AuthenticationService.java]: Can you update to be 2014-2016

File: src/main/java/org/openbmp/db_rest/resources/UserAuth.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
+package org.openbmp.db_rest.resources;
+
+/**
+ * Created by ALEX on 3/18/16.
+ */
+/*
+ * Copyright (c) 2014 Cisco Systems, Inc. and others.  All rights reserved.
```
Comment (TimEvens) [inline src/main/java/org/openbmp/db_rest/resources/UserAuth.java]: change to 2016.

File: src/main/java/org/openbmp/db_rest/helpers/RestAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2014-2015 Cisco Systems, Inc. and others.  All rights reserved.
```
Comment (TimEvens) [inline src/main/java/org/openbmp/db_rest/helpers/RestAuthenticationFilter.java]: change to 2014-2016

**Comment by BinaryDevil** [comment]
OK modified @TimEvens

**Comment by BinaryDevil** [comment]
@TimEvens Please **check and merge**

File: src/main/java/org/openbmp/db_rest/resources/Updates.java [hunk 4]
```diff
@@ -320,9 +320,10 @@ public Response getUpdatesOverTime(@QueryParam("searchPeer") String searchPeer,
                 + interval + ") as IntervalTime,\n");
         query.append("               count(*) as Count\n");
         query.append("      FROM path_attr_log l\n");
-		query.append("      JOIN bgp_peers p ON (l.peer_hash_id = p.hash_id)\n");
-		query.append("      JOIN routers r ON (p.router_hash_id = r.hash_id)\n");
-		query.append("      JOIN collectors c ON (r.collector_hash_id = c.hash_id)\n");        query.append("      WHERE l.timestamp >= "+startTimestamp +" AND l.timestamp <= " + endTimestamp + "\n");
+//		query.append("      JOIN bgp_peers p ON (l.peer_hash_id = p.hash_id)\n");
+//		query.append("      JOIN routers r ON (p.router_hash_id = r.hash_id)\n");
+//		query.append("      JOIN collectors c ON (r.collector_hash_id = c.hash_id)\n");
```
Comment (TimEvens) [inline src/main/java/org/openbmp/db_rest/resources/Updates.java:325]: We can work on adding this back later.... but sometimes nested and STRAIGHT_JOINS are better for this type of query.

**Pull request merged**